packages/components/src/components/rating/rating.jsx

```diff
@@ -0,0 +1,140 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Text from '../text';
+
+const Rating = ({
+    allow_half_rating = false,
+    background_class,
+    direction = 'ltr',
+    disable_hover = false,
+    default_content,
+    element_class,
+    icon_selected,
+    icon_unselected,
+    max_rating = 5,
+    onClick,
+    readonly = false,
+    show_rating = false,
+    value,
+}) => {
+    const [enabled_rate_index, setEnabledRateIndex] = React.useState(-1);
+    const [hovered_rate_index, setHoveredRateIndex] = React.useState(-1);
+    const [is_hover_enabled, setIsHoverEnabled] = React.useState(false);
+    const ratingContainerRef = React.useRef(null);
+
+    const calculateRating = e => {
+        const { width, left, right } = ratingContainerRef.current.getBoundingClientRect();
+        const rating_start_element = direction === 'ltr' ? e.clientX - left : right - e.clientX;
+        const selected_rating = (rating_start_element / width) * max_rating;
+        if (selected_rating > 0.4) {
```

Comment:
as per the suggestion [here](https://github.com/binary-com/deriv-app/pull/5495#discussion_r880046899), we don't need this condition